### PR TITLE
Create initial game endpoints

### PR DESCRIPTION
**Pull request by mathewdenison**

### CODE REVIEW

File: clue_game_backend/api_contract.md [hunk 1]
```diff
@@ -7,7 +7,6 @@ POST /api/game/
 		{
 			"id": "210730a4-6d46-11ee-b962-0242ac120002"
 			"join_code": "pawn123",
-			"session_id": "210730a4-6d46-11ee-b962-0242ac120002", // When "someone" creates a game, the BE is going to create a user for this person, this is their session id
```
Comment (dporth) [inline clue_game_backend/api_contract.md:10]: This session id is actually needed. When a person creates a game, according to the flow we talked about, that person doesn't have a user created in our system yet. We create a user for that person when that person makes a request to create a game. The FE then needs to know the session id of this person.

Comment (dporth) [reply clue_game_backend/api_contract.md:10]: You'll need to create a user in the POST game endpoint.

File: clue_game_backend/clue_game_api/game/views.py [hunk 2]
```diff
@@ -11,17 +11,18 @@
 from django.views.decorators.csrf import csrf_exempt
 from rest_framework.views import APIView
 from .models import Game
+# from clue_game_api.users.models import User
 from .serializers import GameSerializer
 
 
 class GameView(APIView):
     def post(self, request):
         game = Game()
```
Comment (dporth) [inline clue_game_backend/clue_game_api/game/views.py:20]: Where is game join code being set?

Comment (dporth) [inline clue_game_backend/clue_game_api/game/views.py:20]: Where is game join code being set?

File: clue_game_backend/api_contract.md [hunk 3]
```diff
@@ -7,7 +7,6 @@ POST /api/game/
 		{
 			"id": "210730a4-6d46-11ee-b962-0242ac120002"
 			"join_code": "pawn123",
```
Comment (dporth) [inline clue_game_backend/api_contract.md:9]: This session id is actually needed. When a person creates a game, according to the flow we talked about, that person doesn't have a user created in our system yet. We create a user for that person when that person makes a request to create a game. The FE then needs to know the session id of this person.

Comment (dporth) [reply clue_game_backend/api_contract.md:9]: You'll need to create a user in the POST game endpoint.

**Review by dporth — CHANGES_REQUESTED** [review]

File: clue_game_backend/clue_game_api/game/views.py (hunk 2 above)
Comment (mathewdenison) [reply clue_game_backend/clue_game_api/game/views.py:20]: It is being set in the models.py which is calling the create_random_join_code method. It is set as a default any time a new Game object is being created.

File: clue_game_backend/api_contract.md (hunk 3 above)
Comment (mathewdenison) [reply clue_game_backend/api_contract.md:9]: It is commented out until Daniel commits and merges his user endpoint. Don't want to add my own for sake of merge conflicts.

Comment (mathewdenison) [reply clue_game_backend/api_contract.md:9]: As for the session_id, that is also commented out for now, and will be implemented when we have the user endpoints created.

Comment (mathewdenison) [reply clue_game_backend/api_contract.md:9]: I'll add it back in as part of the api contract though. It was discussed at the meeting while you weren't there that we were supposedly planning to get rid of the session_id.

File: clue_game_backend/clue_game_api/game/views.py (hunk 2 above)
Comment (dporth) [reply clue_game_backend/clue_game_api/game/views.py:20]: Nice

**Comment by dporth** [comment]
Overall looks good, can you put in line code comments like "# TODO ..." to make known the things you are waiting on the user's endpoint to be done.

**Comment by mathewdenison** [comment]
Yep I'll add that when I get home tonight along with changing back the apicontract file.

**Comment by mathewdenison** [comment]
Updates made per requested changes. 
1. external_game_id changed to id
2. TODOs added
3. Removed change from api contracts file

**Pull request closed without merging**